extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/taskadapter/PodTemplateTaskAdapter.java

```diff
@@ -212,27 +212,36 @@ public K8sTaskId getTaskId(Job from)
 
   private HashMap<String, PodTemplate> initializePodTemplates(Properties properties)
   {
-    HashMap<String, PodTemplate> podTemplateMap = new HashMap<>();
-    Optional<PodTemplate> basePodTemplate = loadPodTemplate("base", properties);
-    if (!basePodTemplate.isPresent()) {
+    Set<String> taskAdapterTemplateKeys = getTaskAdapterTemplates(properties);
+    if (!taskAdapterTemplateKeys.contains("base")) {
       throw new IAE("Pod template task adapter requires a base pod template to be specified");
     }
-    podTemplateMap.put("base", basePodTemplate.get());
 
-    MapperConfig config = mapper.getDeserializationConfig();
-    AnnotatedClass cls = AnnotatedClassResolver.resolveWithoutSuperTypes(config, Task.class);
-    Collection<NamedType> taskSubtypes = mapper.getSubtypeResolver().collectAndResolveSubtypesByClass(config, cls);
-    for (NamedType namedType : taskSubtypes) {
-      String taskType = namedType.getName();
-      Optional<PodTemplate> template = loadPodTemplate(taskType, properties);
-      template.ifPresent(podTemplate -> podTemplateMap.put(taskType, podTemplate));
+    HashMap<String, PodTemplate> podTemplateMap = new HashMap<>();
+    for (String taskAdapterTemplateKey : taskAdapterTemplateKeys) {
+      Optional<PodTemplate> template = loadPodTemplate(taskAdapterTemplateKey, properties);
+      template.ifPresent(podTemplate -> podTemplateMap.put(taskAdapterTemplateKey, podTemplate));
     }
     return podTemplateMap;
   }
 
+  private static Set<String> getTaskAdapterTemplates(Properties properties)
+  {
+    Set<String> taskAdapterTemplates = new HashSet<>();
+
+    for (String runtimeProperty : properties.stringPropertyNames()) {
+      if (runtimeProperty.startsWith(TASK_PROPERTY)) {
+        String[] taskAdapterPropertyPaths = runtimeProperty.split("\\.");
+        taskAdapterTemplates.add(taskAdapterPropertyPaths[taskAdapterPropertyPaths.length - 1]);
+      }
+    }
+
+    return taskAdapterTemplates;
+  }
+
   private Optional<PodTemplate> loadPodTemplate(String key, Properties properties)
   {
-    String property = StringUtils.format(TASK_PROPERTY, key);
+    String property = TASK_PROPERTY + key;
     String podTemplateFile = properties.getProperty(property);
     if (podTemplateFile == null) {
       log.debug("Pod template file not specified for [%s]", key);
```

Comment:
nit: this feels like it should be a warn at least or maybe even throw an exception here. Is there a reason we expect podTemplateFile to be null?

Reply:
Also since you are making changes here, on line 254 - should the error be an `IAE` or something else to indicate the user passed in a filename that Druid can't read successfully.

Reply:
actually i think it should technically never be null now since we call it by listing props first, it seems fine to throw an exception